### DISCUSSION TITLE
Ruleset validator

### DISCUSSION THREAD

**Discussion by matthias-ronge** [kitodo/kitodo-production#6168] [Ideas]
Da wir im Frontend noch keinen Modus zum Bearbeiten von Regelsätzen haben, komme ich immer wieder auf die Idee, dass wir vielleicht zumindest eine Art Regelsatzvalidator bauen könnten, der zumindest auf semantische Inkonsistenzen hinweisen kann. Zum Beispiel, wenn eine `<restriction>` einen `<permit key="…"/>` enthält, der nicht definiert ist, wenn eine `<division>` zwei mal mit derselben `id` definiert ist, und solche Dinge. Ich lasse das hier als Idee stehen und möchte insbesondere fragen, welche die typischsten Fehler beim Bearbeiten eines Regelsatzes sind, die manchen in Erinnerung geblieben sind, und die so ein Modul vielleicht überprüften sollte?

**Comment by henning-gerhardt** [comment]
Warum wird nicht fürs Validieren der Rulesets / Regelsätze die [ruleset.xsd](https://github.com/kitodo/kitodo-production/blob/master/Kitodo/rulesets/ruleset.xsd) grundlegend genutzt? Darauf aufbauend kann man ja noch weitere Regeln via bspw. Schematron oder was ähnlichen dazu nehmen, dass auf noch (mehr) logische Sachen prüft, die via XSD nicht oder nur sehr schwer umsetzbar sind?
